### PR TITLE
Top reconstruction and data-driven background estimation from WZCR/TTCR.

### PR DESCRIPTION
**Pull request by yhlee52**
Top kinematic reconstruction : fcncKinematicReco.py
Bkg estimation : bkgest_working.py
1. Top kin reco
   - Reconstruction algorithm is validated.
   - Top kin reco postprocessor only save two top events. : Now this command is commented out.
2. Bkg estimation
   - This code is working but fit result does not look good.
   - Now the code makes few png files for visualize fit result and one txt file with estimated result.
   - Change plot cosmetics and get entries from fitting function.
   - Suggested output : Save the result as .yaml format and use it for new control plots.
3. Missed LeptonSF
   - LeptonSF is added on the plot drawing step.

### CODE REVIEW

File: TopAnalysis/data/histogramming/fcncTrilepton.yaml [hunk 1]
```diff
@@ -107,7 +107,23 @@ steps:
               "nGoodJet", "nBjet", "nGoodJet_nBjet", "GoodJet_CSVv2",
               "MET_pt", "MET_phi",]
 
+    - name: "WZCR_0b"
+      cuts: ["CutStep >= 2", "nGoodJet>=1", "nGoodJet<=3", "W_MT <= 300", "TMath::Abs(Z_mass-91.2)<7.5", "nBjet==0"]
+      hists: ["nPV", "nPV_noWeight",
+              "Lepton1_pt", "Lepton2_pt", "Lepton3_pt",
+              "Lepton1_eta", "TriLepton_mass", "TriLepton_pt", "TriLepton_WleptonZdPhi", "TriLepton_WleptonZdR",
```
Comment (jhgoh) [inline TopAnalysis/data/histogramming/fcncTrilepton.yaml]: add Lepton2_eta Lepton3_eta

File: TopAnalysis/python/postprocessing/fcncKinematicReco.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        self.out = wrappedOutputTree
+        self.out.branch("SMTop_pt", "F")
+        self.out.branch("SMTop_eta", "F")
+        self.out.branch("SMTop_phi", "F")
+        self.out.branch("SMTop_mass", "F")
+        self.out.branch("FCNCTop_pt", "F")
+        self.out.branch("FCNCTop_eta", "F")
+        self.out.branch("FCNCTop_phi", "F")
+        self.out.branch("FCNCTop_mass", "F")
+        self.out.branch("Reconstructed", "i")
```
Comment (jhgoh) [inline TopAnalysis/python/postprocessing/fcncKinematicReco.py]: I prefer
"KinTopWb_*"
"KinTopZq_*"
"KinTop_status"

File: TopAnalysis/python/postprocessing/fcncKinematicReco.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+    def analyze(self, event):
+        Wmass = 80.4 # Fixed W mass
+        OriginTmass = 172.5 # Fixed T mass
+
+        ## initialize ##
+        SMTKinVal = []
+        FCNCKinVal = []
+
+        if ( event._tree.b_out_GoodLeptonCode != 111 ): return False
```
Comment (jhgoh) [inline TopAnalysis/python/postprocessing/fcncKinematicReco.py]:
```
SMTKinVal = [0,0,0,0]
FCNCTKinVal = [0,0,0,0]
consted = 0

while True:
    if ( event._tree.b_out_GoodLeptonCode != 111 ): break
    if ( ..... ): break

    ....

self.out.fillBranch("SMTop_pt", SMTKinVal[0])
self.out.fillBranch("SMTop_eta", SMTKinVal[1])
self.out.fillBranch("SMTop_phi", SMTKinVal[2])
self.out.fillBranch("SMTop_mass", SMTKinVal[3])
self.out.fillBranch("FCNCTop_pt", FCNCTKinVal[0])
self.out.fillBranch("FCNCTop_eta", FCNCTKinVal[1])
self.out.fillBranch("FCNCTop_phi", FCNCTKinVal[2])
self.out.fillBranch("FCNCTop_mass", FCNCTKinVal[3])

File: TopAnalysis/data/histogramming/fcncTrilepton.yaml (hunk 1 above)
Comment (yhlee52) [reply TopAnalysis/data/histogramming/fcncTrilepton.yaml]: The comment is applied.

File: TopAnalysis/test/fcncTriLepton/bkgestFit_working.py [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    linmodel.fitTo(TTdata, RooFit.Range(110, 150))
+    coef1.Print()
+    linmodel.plotOn(tframe)
+    tframe.Draw()
+    t2 = coef1.getVal()
+    ttplot.SaveAs("fitresult_TTCR_%s.png"%ch)
+    t11 = hTT_all.Integral(30, 70)
+    t22 = hTT_all.Integral(110, 150) 
+    file = open(result, "a")
+    tline = "%s TT %f %f\n"%(ch, 15*(abs(t1)+abs(t2)), 35900*nttsr)
```
Comment (jhgoh) [inline TopAnalysis/test/fcncTriLepton/bkgestFit_working.py]: 35900 은 무슨 숫자인가요

Comment (yhlee52) [reply TopAnalysis/test/fcncTriLepton/bkgestFit_working.py]: 엔트리 수 구할때 곱해주는 luminosity 입니다.

File: TopAnalysis/python/postprocessing/fcncKinematicReco.py (hunk 2 above)
Comment (yhlee52) [reply TopAnalysis/python/postprocessing/fcncKinematicReco.py]: This comment is applied on code.

File: TopAnalysis/python/postprocessing/fcncKinematicReco.py (hunk 3 above)
Comment (yhlee52) [reply TopAnalysis/python/postprocessing/fcncKinematicReco.py]: This comment is applied on code.

**Pull request merged**